hudi-spark/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -142,6 +143,16 @@ object DataSourceWriteOptions {
   val MOR_TABLE_TYPE_OPT_VAL = HoodieTableType.MERGE_ON_READ.name
   val DEFAULT_TABLE_TYPE_OPT_VAL = COW_TABLE_TYPE_OPT_VAL
 
+  /**
+    * The table base file format for the underlying data, for this write.
+    * Note that this can't change across writes.
+    *
+    * Default: PARQUET
+    */
+  val TABLE_FILE_FORMAT_OPT_KEY = "hoodie.table.base.file.format"
```

Comment:
sorry did not catch this before.. so can we introduce this at the `WriteConfig` level, that way it's common for both datasource and deltastreamer.. 
Any write config can be passed through the datasource anyway

Reply:
Done, thanks